### PR TITLE
Shipping Labels: view order summary and Woo discount p1

### PR DESCRIPTION
**Pull request by hichamboushaba**
First part of #3617, this just adds the layout of the order summary, and the logic to display it. But as #3525 is not done yet, we can't have a real calculation of the order price and discount.

|  |  |
| --- | --- |
|Screenshot_1614175365|Screenshot_1614175367|
|Screenshot_1614176182| |

#### Testing
1. Change this [line](https://github.com/woocommerce/woocommerce-android/blob/eb824a2d966f4f20160fd35099e92485e5551fba/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt#L157) to: 
``` kotlin
      is SideEffect.ShowCarrierOptions -> handleResult { Event.ShippingCarrierSelected }
```
to allow completing the shipping carriers step.
2. Open details of an order that satisfies the shipping label creation.
3. Click on "create shipping label"
4. Confirm that the order summary is hidden.
5. Pass all steps.
6. Confirm that the order summary is displayed after reaching the payment step.
7. Click on the ℹ️ image, and confirm that the discount bottom sheet is displayed.

To test the other scenario: order without discount, use `BigDecimal.ZERO` for the discount field here: https://github.com/woocommerce/woocommerce-android/blob/a9b4af2f1aa93b37c340a7d591e0861bee928887/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt#L288-L290

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt [hunk 1]
```diff
@@ -212,7 +214,8 @@ class CreateShippingLabelViewModel @AssistedInject constructor(
                     packagingDetailsStep = Step.notDone(),
                     customsStep = Step.notDone(),
                     carrierStep = Step.notDone(),
-                    paymentStep = Step.notDone(data.currentPaymentMethod.stepDescription)
+                    paymentStep = Step.notDone(data.currentPaymentMethod.stepDescription),
+                    orderSummaryState = OrderSummaryState()
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt:218]: This is nothing, just noticed that the parameter already has the same default value.

**Review by 0nko — COMMENTED** [review]
Looks beautiful! 

I know this is unrelated, but when I was testing the PR I managed to crash the app when I tapped on the Done button in the Payments (the payment list is empty). I'm not sure if it's fully implemented, just thought I'd mention it.

image

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt:218]: Yes, but we need to do it, and not rely on the default value, as the steps are not always ordered, if the user for example did all the steps, we will show the order summary, but if they update the list of packages, we need to invalidate the rates, and hide the order summary.
That's something I was planning to discuss with you regarding how we should handle it in the state machine, I'll ping you in Slack 🙂

**Comment by hichamboushaba** [comment]
> Looks beautiful!
> 
> I know this is unrelated, but when I was testing the PR I managed to crash the app when I tapped on the Done button in the Payments (the payment list is empty). I'm not sure if it's fully implemented, just thought I'd mention it.
> 
> image

That's expected, as the payment step on its current state expects that the user has at least one payment method, after implementing the eligibility API endpoint, the button "create shipping label" should be hidden unless there is one payment method.

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**